src/transport.c

```diff
@@ -108,10 +108,13 @@ static int find_address(unsigned char ipaddr[BOOTH_IPADDR_LEN],
 				break;
 
 		if (matched == node->addrlen) {
-			*address_bits_matched = matched * 8;
-			*me = node;
-			did_match = EXACT_MATCH;
-			break;
+			if((matched_tmp < matched * 8)||((node->type == SITE)&&(matched_tmp == matched * 8)))
```

Comment:
Hi Dejan,  @krig encountered this issue that "booth revoke <ticket>"  fails:

ha2:~ # booth list
ticket: ticketA, leader: 10.12.2.201, expires: 2016-08-23 08:29:25
ticket: ticketB, leader: NONE
ha2:~ # booth revoke ticketA
Aug 23 08:19:51 ha2 booth: [5610]: ERROR: We're just an arbitrator, cannot grant/revoke tickets here.

But ha2 is not the arbitrator, it is a node in the cluster currently holding the ticket.

Configuration file:

# The booth configuration file is "/etc/booth/booth.conf". You need to
# prepare the same booth configuration file on each arbitrator and
# each node in the cluster sites where the booth daemon can be launched.
# Here is an example of the configuration file:

# "transport" means which transport layer booth daemon will use.
# Currently only "UDP" is supported.
transport="UDP"

# The port that booth daemons will use to talk to each other.
port="9929"

# The arbitrator IP. If you want to configure several arbitrators,
# you need to configure each arbitrator with a separate line.
arbitrator="10.12.2.105"

# The site IP. The cluster site uses this IP to talk to other sites.
# Like arbitrator, you need to configure each site with a separate line.
site="10.12.2.201"
site="10.12.2.202"

# The ticket name, which corresponds to a set of resources which can be
# fail-overed among different sites.
ticket="ticketA"
ticket="ticketB"
	expire = 600
	weights = 1,2,3